### PR TITLE
Fix Prometheus and Alertmanager templates to support routePrefix

### PR DESCRIPTION
**Pull request by gmichels**
Fix Prometheus and Alertmanager templates to support routePrefix.

Credit goes to @kovszilard as pretty much all information was taken from https://github.com/coreos/prometheus-operator/issues/1596

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

**Comment by vsliouniaev** [comment]
Hey @gmichels could you fix the sign-off on `68cf08a` ?

File: stable/prometheus-operator/templates/alertmanager/ingress.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if and .Values.alertmanager.enabled .Values.alertmanager.ingress.enabled }}
-{{- $routePrefix := .Values.routePrefix }}
+{{- $routePrefix := .Values.alertmanager.alertmanagerSpec.routePrefix }}
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/alertmanager/ingress.yaml:2]: Would you mind applying a similar fix in  `/templates/prometheus/ingress.yaml` ?

File: stable/prometheus-operator/templates/alertmanager/servicemonitor.yaml [hunk 2]
```diff
@@ -17,4 +17,9 @@ spec:
   endpoints:
   - port: web
     interval: 30s
+    {{- if hasSuffix "/" .Values.alertmanager.alertmanagerSpec.routePrefix }}
+    path: "{{ .Values.alertmanager.alertmanagerSpec.routePrefix }}metrics"
+    {{- else }}
+    path: "{{ .Values.alertmanager.alertmanagerSpec.routePrefix }}/metrics"
+    {{- end }}
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/alertmanager/servicemonitor.yaml]: You could turn this into a one-liner by leveraging `trimSuffix`

File: stable/prometheus-operator/templates/prometheus/servicemonitor.yaml [hunk 3]
```diff
@@ -17,4 +17,9 @@ spec:
   endpoints:
   - port: web
     interval: 30s
+    {{- if hasSuffix "/" .Values.prometheus.prometheusSpec.routePrefix }}
+    path: "{{ .Values.prometheus.prometheusSpec.routePrefix }}metrics"
+    {{- else }}
+    path: "{{ .Values.prometheus.prometheusSpec.routePrefix }}/metrics"
+    {{- end }}
```
Comment (vsliouniaev) [inline stable/prometheus-operator/templates/prometheus/servicemonitor.yaml]: Same as above, you could use `trimSuffix` here

**Review by vsliouniaev — CHANGES_REQUESTED** [review]
Alertmanager part works
- can access alertmanager only on specified prefix
- grafana works
- prometheus sends alerts to and scrapes alertmanager

Prometheus needs a change to its ingress configuration to also apply the prefix

File: stable/prometheus-operator/templates/alertmanager/ingress.yaml (hunk 1 above)
Comment (gmichels) [reply stable/prometheus-operator/templates/alertmanager/ingress.yaml:2]: Thanks for catching this. For some odd reason it was working when going to my ingress with /prometheus, although I had noticed the lack of the `path` property in the ingress definition.

File: stable/prometheus-operator/templates/alertmanager/servicemonitor.yaml (hunk 2 above)
Comment (gmichels) [reply stable/prometheus-operator/templates/alertmanager/servicemonitor.yaml]: Done, thanks for the suggestion.

File: stable/prometheus-operator/templates/prometheus/servicemonitor.yaml (hunk 3 above)
Comment (gmichels) [reply stable/prometheus-operator/templates/prometheus/servicemonitor.yaml]: Done, thanks for the suggestion.

File: stable/prometheus-operator/templates/alertmanager/ingress.yaml (hunk 1 above)
Comment (vsliouniaev) [reply stable/prometheus-operator/templates/alertmanager/ingress.yaml:2]: It also worked when I tested it with the Istio configuration we have but it seemed odd. Didn't work on pure ingress though

**Comment by vsliouniaev** [comment]
Looks like this is working on my machine.

Could you fix the sign-off on `68cf08a8d1ed788ec3258ba0b7b73069de6dedff` and update the chart version again? (there was a merge while your PR was open)

**Comment by gmichels** [comment]
I am trying to find a way to sign-off on a single commit, but all I can find is a way to sign-off on all of my commits, which would result in double sign-offs on all commits except `68cf08a8d1ed788ec3258ba0b7b73069de6dedff`.

Do you have a command handy to sign-off on a specific previous commit? Or if you are good with me double signing-off on all commits, that will work as well.

**Comment by vsliouniaev** [comment]
No matter what you do all commits will be rewritten and history will be lost so you can squash all your commits into one.
A double sign-off will not have ill effects.
You can use interactive rebase to do so `git rebase origin/master -i` and then add the line to the single commit.

**Comment by gmichels** [comment]
Rebased, sign-off added and chart version bumped.

**Comment by vsliouniaev** [comment]
Please fix CI

**Comment by gmichels** [comment]
Sorry, first PR ever here, so I am learning as I go.

**Comment by vsliouniaev** [comment]
@gmichels if you find me (vas) on kubernetes.slack.com I can help you out

**Comment by gmichels** [comment]
@vsliouniaev thanks for the help, will do that on Monday. Or if you feel it would be better, I can simply close this and submit a fresh PR.

**Comment by gianrubio** [comment]
@gmichels please fix CI

**Comment by gmichels** [comment]
@gianrubio fixed, thanks

**Comment by gmichels** [comment]
Hey @gianrubio, just wondering if there is anything left for me to do to get this merged? Thank you

**Pull request merged**